internal/host/installcmd.go

```diff
@@ -95,6 +95,11 @@ func (i *installCmd) GetStep(ctx context.Context, host *models.Host) (*models.St
 		}
 	}
 
+	if i.instructionConfig.ServiceCACertPath != "" {
```

Comment:
This is still strange to me. It looks like we never use the value in `i.instructionConfig.ServiceCACertPath`.
I guess it's just more convenient to use the existence of the `SERVICE_CA_CERT_PATH` env var in combination with `envconfig` to achieve this bool-type behavior?

Reply:
The only place where the value of ServiceCACertPath is used is to read the contents and put it into HostCACertPath, and that happens during the discovery iso ignition. The usage here is to achieve the bool-type behavior to generate the right instruction for the host. We could introduce a new env var called ConfigureServiceCACert like you said, but the presence or absense of ServiceCACertPath does the same and does it better, I think. Like there is no reason for ServiceCACertPath to be set to a value and ConfigureServiceCACert to be set to false.

Reply:
Right, so it's the same env var, but we're not using the value of the `ServiceCACertPath` field of `InstructionConfig` anywhere.

I was hoping we could make the field unexported then make a method called `WriteCACert` or something but it looks like `envconfig` doesn't work with unexported fields. I'll leave it up to you if you want to create the method if you think it makes things easier to understand.

Reply:
After looking at the job vs local job distinction for the other conversation I found this [flag that `main` uses](https://github.com/openshift/assisted-service/blob/0cfb84ab9a5722cc8c86efe96f2580fc433b9318/cmd/main.go#L174) to distinguish if it is running on-prem or in the kube. Maybe we should use that instead of the path in this case, it would probably also be useful for the `/etc/hosts` case.

Reply:
ServiceCACert is not specific to onprem only though. I expect the cloud version to use the same code as is or with minimal changes. The only difference would be how the assisted-service itself is configured with SSL (in our case, we are doing it with ngnix fronting the service), but how discovery ignition, master/work and controller flow should be the same AFAIK.

Reply:
Added `hasCACert` method.